File: backend/app/models/message.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.database import Base


class Message(Base):
    __tablename__ = "messages"

    id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        index=True,
    )

    customer_name: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )

    email: Mapped[str] = mapped_column(
        String(150),
        nullable=False,
    )

    text: Mapped[str] = mapped_column(
        Text,
        nullable=False,
    )

    status: Mapped[str] = mapped_column(
        String(20),
        default="pending",
        nullable=False,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
    )

    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
    )

    reviewed_at: Mapped[datetime | None] = mapped_column(
        DateTime,
        nullable=True,
    )

    analysis = relationship(
        "Analysis",
        back_populates="message",
        uselist=False,
        cascade="all, delete-orphan",
    )

File: frontend/src/types/message.ts
export interface Message {
  id: number;
  customer_name: string;
  email: string;
  text: string;
  status: string;
  created_at: string;
  updated_at: string;
  reviewed_at: string | null;
}

export interface Analysis {
  id: number;
  message_id: number;
  summary: string;
  category: string;
  priority: string;
  confidence: number;
  suggested_reply: string;
  created_at: string;
  updated_at: string;
}